functions/Set-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,332 @@
+
+.PARAMETER OutputFileName
+The name of the file in which the output of this step is saved.
+
+.PARAMETER ProxyName
+The name of the proxy that the job step runs as.
+
+.NOTES 
+Original Author: Sander Stad (@sqlstad, sqlstad.nl)
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Set-DbaAgentJobStep
+
+.EXAMPLE   
+Set-DbaAgentJobStep -SqlServer 'sql' -JobName 'Job1' -StepName 'Step1' -NewName 'Step2'
+Changes the name of the step in "Job1" with the name "Step1" to "Step2"
+
+.EXAMPLE   
+Set-DbaAgentJobStep -SqlServer 'sql' -JobName 'Job1' -StepName 'Step1' -Database 'msdb'
+Changes the database of the step in "Job1" with the name "Step1" to 'msdb'
+
+.EXAMPLE   
+"sql1", "sql2", "sql3" | Set-DbaAgentJobStep -JobName 'Job1' -StepName 'Step1' -Database 'msdb'
+Changes the database of the step in "Job1" with the name "Step1" to 'msdb' for multiple servers
+
+#>   
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+
+    param(
+        [parameter(Mandatory = $true, ValueFromPipeline=$true)]
+		[object[]]$SqlServer,
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$JobName,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$StepName,
+        [Parameter(Mandatory = $false)]
+        [string]$NewName,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('ActiveScripting','AnalysisCommand','AnalysisQuery','CmdExec','Distribution','LogReader','Merge','PowerShell','QueueReader','Snapshot','Ssis','TransactSql')]
+        [string]$Subsystem,
+        [Parameter(Mandatory = $false)]
+        [string]$Command,
+        [Parameter(Mandatory = $false)]
+        [int]$CmdExecSuccessCode,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('QuitWithSuccess', 'QuitWithFailure', 'GoToNextStep', 'GoToStep')]
+        [string]$OnSuccessAction,
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($OnSuccessAction -ne 'GoToStep') -and ($_ -ge 1))
+            {
+                Throw "Parameter OnSuccessStepId can only be used with OnSuccessAction 'GoToStep'."
```

Comment:
Either ignore in that case or handle in begin block with the more smooth `Stop-Function` handling.
If we choose to keep the interrupt, the comment based help should reflect this action.